### PR TITLE
fix: consider the referenced nodes in `traverseNodes`

### PR DESCRIPTION
**Pull request by cool-develope**
Closes: #718

### CODE REVIEW

**Comment by cool-develope** [comment]
@catShaark
I cleaned the code based on your PR, thanks for contributing again

**Comment by tac0turtle** [comment]
which pr do we keep open? this or catsharks?

**Comment by cool-develope** [comment]
this PR include some extra cleaning ups, let's move with @catShaark firstly

File: nodedb.go [hunk 1]
```diff
@@ -511,6 +511,15 @@ func (ndb *nodeDB) HasVersion(version int64) (bool, error) {
 	return ndb.db.Has(nodeKeyFormat.Key(version, []byte{1}))
 }
 
+func isReferenceToRoot(bz []byte) bool {
+	if bz[0] == nodeKeyFormat.Prefix()[0] {
+		if len(bz) == 13 {
```
Comment (yihuang) [inline nodedb.go:516]: It seems the condition for a referenced node is hacky, maybe we should add a prefix to normal node values as well, so we can distinguish with prefix alone.

**Pull request closed without merging**